src/DynamoCoreWpf/Views/Preview/Watch3DView.xaml.cs

```diff
@@ -857,267 +902,153 @@ public void RenderDrawables(VisualizationEventArgs e)
+                        parameters.Text.TextInfo.Add(new TextInfo(HelixRenderPackage.CleanTag(rp.Description), new Vector3(pt.X + 0.025f, pt.Y + 0.025f, pt.Z + 0.025f)));
+                    }
 
-            if (color_idx <= p.TriangleVertexColors.Count-3)
-            {
-                color = new Color4(
-                (float)(p.TriangleVertexColors[color_idx] / 255.0),
-                (float)(p.TriangleVertexColors[color_idx + 1] / 255.0),
-                (float)(p.TriangleVertexColors[color_idx + 2] / 255.0),
-                (float)(p.TriangleVertexColors[color_idx + 3] / 255.0));
+                    MeshCount++;
+                }
             }
-           
-            return color;
         }
 
-        private static Vector3 GetVertex(List<double> p, int i)
+        private void SendGraphicsToView(GraphicsUpdateParameters parameters)
         {
-            var x = (float)p[i];
-            var y = (float)p[i + 1];
-            var z = (float)p[i + 2];
+            Points = parameters.Points;
+            Lines = parameters.Lines;
+            LinesSelected = parameters.SelectedLines;
+            Mesh = parameters.Mesh;
+            MeshSelected = parameters.SelectedMesh;
+            PerVertexMesh = parameters.PerVertexMesh;
+            Text = parameters.Text;
 
-            // DirectX convention - Y Up
-            var new_point = new Vector3(x, z, -y);
-            return new_point;
+            // Send property changed notifications for everything
+            NotifyPropertyChanged(string.Empty);
         }
 
-        private string CleanTag(string tag)
-        {
-            var splits = tag.Split(':');
-            if (splits.Count() <= 1) return tag;
+        #endregion
+    }
 
-            var sb = new StringBuilder();
-            for (int i = 1; i < splits.Count(); i++)
-            {
-                sb.AppendFormat("[{0}]", splits[i]);
-            }
-            return sb.ToString();
-        }
+    internal class GraphicsUpdateParameters
+    {
+        public PointGeometry3D Points { get; set; }
+        public LineGeometry3D Lines { get; set; }
+        public LineGeometry3D SelectedLines { get; set; }
+        public MeshGeometry3D Mesh { get; set; }
+        public MeshGeometry3D SelectedMesh { get; set; }
+        public MeshGeometry3D PerVertexMesh { get; set; }
+        public BillboardText3D Text { get; set; }
+    }
 
-        #endregion
+    internal class PackageAggregationParameters
+    {
+        public IEnumerable<HelixRenderPackage> Packages { get; set; } 
+        public PointGeometry3D Points { get; set; }
+        public LineGeometry3D Lines { get; set; }
+        public LineGeometry3D SelectedLines { get; set; }
+        public MeshGeometry3D Mesh { get; set; }
+        public MeshGeometry3D PerVertexMesh { get; set; }
+        public MeshGeometry3D SelectedMesh { get; set; }
+        public BillboardText3D Text { get; set; }
     }
```

Comment:
Small bone to pick, but I find the `Params` suffix to be much shorter and just as readable.